lib/presentation/screens/ship_it/components/contact_share_card.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+
+import 'package:chalk_out/presentation/blocs/ship_it/ship_it_bloc.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ContactShareCard extends StatelessWidget {
+  const ContactShareCard({
+    Key key,
+    @required this.totalRequiredPlayersBool,
+    @required this.contactName,
+    @required this.index,
+  }) : super(key: key);
+  final String contactName;
+  final int index;
+  final bool totalRequiredPlayersBool;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      // width: MediaQuery.of(context).size.width,
+      padding: const EdgeInsets.only(top: 16),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisSize: MainAxisSize.max,
+        children: [
+          Text(
+            index.toString(),
+            style: TextStyle(fontSize: 30, color: (totalRequiredPlayersBool) ? Colors.black : Colors.red),
+          ),
+          Padding(
+            padding: const EdgeInsets.only(top: 4.0),
+            child: Text(
+              '/10',
+            ),
+          ),
+          Expanded(
+            child: Row(
+              mainAxisSize: MainAxisSize.max,
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Padding(
+                  padding: EdgeInsets.only(left: 8.0, bottom: contactName.isEmpty ? 0 : 8),
+                  child: (contactName.isEmpty)
+                      ? RaisedButton(
+                          onPressed: () {
+                            context.bloc<ShipItBloc>()
+                              ..add(
+                                AddPlayerStarted(),
+                              );
+                          },
+                          child: Text('Add Player'),
+                        )
+                      : Container(
+                          child: Text(contactName),
+                          decoration: BoxDecoration(
+                            border: Border(
+                              bottom: BorderSide(color: Colors.black),
+                            ),
+                          ),
+                        ),
+                ),
+                (contactName.isEmpty)
+                    ? Container()
+                    : IconButton(
+                        icon: Icon(
+                          Icons.delete,
+                          color: Colors.red,
+                        ),
+                        onPressed: () {
+                          context.bloc<ShipItBloc>()..add(RemovePlayerPressed(index: index));
+                        },
```

Comment:
you'd be fine with just 1 dot

Reply:
The one reason that you would do the 2 dots is so that you can return the bloc instance AND perform an event all at once. But because you don't need to return the instance, you'd be fine without the 2 dots here.